### PR TITLE
fix and test FrequencyModulatedArtifactWriter

### PR DESCRIPTION
**Pull request by dbkeitel**
@Rodrigo-Tenorio followup to #277

 - needed to adapt concatenate_sft_files()
   method to changed lalapps_splitSFTs filename convention

### CODE REVIEW

**Comment by dbkeitel** [comment]
...now with less actions ;)

File: pyfstat/make_sfts.py [hunk 1]
```diff
@@ -1378,39 +1378,40 @@ def get_h0(self, t):
...
-            self.detectors[1],
-            self.nsfts,
-            self.Tsft,
-            int(self.tstart),
-            int(self.duration),
-            self.label,
+        """Merges the individual SFT files via lalapps_splitSFTs executable."""
+
+        SFTFilename = (
+            f"{self.detectors[0]}-{self.nsfts}_{self.detectors}_{self.Tsft}SFT"
```
Comment (Rodrigo-Tenorio) [inline pyfstat/make_sfts.py:1384]: This line is just beautiful.

I guess `self.detectors` is checked to be a string somewhere?

**Review by Rodrigo-Tenorio — CHANGES_REQUESTED** [review]

File: pyfstat/make_sfts.py (hunk 1 above)
Comment (dbkeitel) [reply pyfstat/make_sfts.py:1384]: > This line is just beautiful.

For reference, the original lalapps snprintf beauty is https://git.ligo.org/lscsoft/lalsuite/-/blob/master/lalapps/src/pulsar/SFTTools/splitSFTs.c#L688

> I guess self.detectors is checked to be a string somewhere?

Purely coincidentally and implicitly, yes:

https://github.com/PyFstat/PyFstat/blob/af6d263c1a8da07e54b3a7649a7fdca855cd2341/pyfstat/make_sfts.py#L1337

**Review by Rodrigo-Tenorio — APPROVED** [review]

File: pyfstat/make_sfts.py (hunk 1 above)
Comment (dbkeitel) [reply pyfstat/make_sfts.py:1384]: But actually if I change in the test to `detectors="H1,L1"`, this one does not catch it. Hmm.

Comment (dbkeitel) [reply pyfstat/make_sfts.py:1384]: Ah, because it's only reached if adding fake Alpha,Delta dependent modulation.

Will add an explicit check earlier.

Comment (dbkeitel) [reply pyfstat/make_sfts.py:1390]: done in d1443ddf0e106ca73ca6b44205e52452507635da

**Review by Rodrigo-Tenorio — APPROVED** [review]

**Pull request merged**